src/main/java/com/atlassian/db/replica/api/context/QueryContext.java

```diff
@@ -0,0 +1,87 @@
+package com.atlassian.db.replica.api.context;
+
+
+import java.util.Objects;
+
+public final class QueryContext {
+    private final Reason reason;
+    private final String sql;
+    private final boolean runOnMain;
+
+
+    private QueryContext(Reason reason, String sql, boolean runOnMain) {
+        this.reason = reason;
+        this.runOnMain = runOnMain;
+        this.sql = sql;
+    }
+
+    public Reason getReason() {
+        return reason;
+    }
+
+    public String getSql() {
+        return sql;
+    }
+
+    public boolean isRunOnMain() {
+        return runOnMain;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        QueryContext that = (QueryContext) o;
+        return runOnMain == that.runOnMain && Objects.equals(reason, that.reason) && Objects.equals(sql, that.sql);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(reason, sql, runOnMain);
+    }
+
+    @Override
+    public String toString() {
+        return "QueryContext{" +
+            "reason=" + reason +
+            ", sql='" + sql + '\'' +
+            ", runOnMain=" + runOnMain +
+            '}';
+    }
+
+    public static QueryContext.Builder builder(Reason reason, boolean runOnMain) {
```

Comment:
What is the reason if we run on replica?

Reply:
READ_OPERATION -> https://github.com/atlassian-labs/db-replica/pull/21/files#diff-5eca8f926943fd2aca9f64d64f90d2d14ed51191273382f5edd4812e49e3d2abR645